packages/cache/src/internal/tar.ts

```diff
@@ -142,11 +147,18 @@ async function getArgs(
     tarPath.type === ArchiveToolType.BSD &&
     compressionMethod !== CompressionMethod.Gzip &&
     IS_WINDOWS
+
   if (BSD_TAR_ZSTD && type !== 'create') {
-    return [...compressionArgs, ...tarArgs].join(' ')
+    args = [[...compressionArgs].join(' '), [...tarArgs].join(' ')]
   } else {
-    return [...tarArgs, ...compressionArgs].join(' ')
+    args = [[...tarArgs].join(' '), [...compressionArgs].join(' ')]
   }
+
+  if (BSD_TAR_ZSTD) {
+    return args
+  }
```

Comment:
Instead of this, just return directly from L152 and L154

Reply:
It returns only in case of bsd tar. Else it will go down